### PR TITLE
41_adface maxsev utility@devel

### PR DESCRIPTION
**Pull request by arjoon-r**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page.
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue so that it closes after successful merging.
- [ ] Address all merge conflicts and resolve appropriately.
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by arjoon-r** [comment]
Hi @bms63, I have already updated the new code for maximum severity with examples #41. Actually ,You have reviewed the (derive_param_maxsev_draft.R )code from a PR which we have created for demo purpose. Sorry for that.  
This is the updated one. Please take a review :)

**Comment by bms63** [comment]
This will fix man pages failing action and probably the lintr fail as well

roxygen2::roxygenize('.', roclets = c('rd', 'collate', 'namespace'))

**Comment by arjoon-r** [comment]
Hi @bms63, All checks are successful. And I have just updated the derive_param_maxsev_draft.R file with New code. So you can review the updated code and give your comments. Since i have modified the draft code and it's not a draft one anymore, Is it possible to rename the .R file in devel?

File: R/derive_param_maxsev_draft.R [hunk 1]
```diff
@@ -1,88 +1,206 @@
...
+#' maximum severity records for the specified variables in `by_vars`. `DTYPE`
+#' will be populated as `MAXIMUM`. `FATEST`will be populated as specified in
+#' `test_maxsev`and `FATESCD` will be populated as specified in `testcd_maxsev`.
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#' library(admiral)
+#' library(dplyr)
```
Comment (bms63) [inline R/derive_param_maxsev_draft.R:64]: Please use the der_adxx for @keywords and @family

File: R/derive_param_maxsev_draft.R [hunk 2]
```diff
@@ -1,88 +1,206 @@
...
 #'
-#'
 derive_param_maxsev <- function(dataset = NULL,
-                                testcd_sev = "SEV",
-                                by_vars = NULL) {
+                                exclude_events = NULL,
+                                filter_sev = "SEV",
+                                test_maxsev = "Maximum Severity",
+                                testcd_maxsev = "MAXSEV",
+                                by_vars = vars(USUBJID, FAOBJ, ATPTREF)) {
```
Comment (bms63) [inline R/derive_param_maxsev_draft.R:130]:
```suggestion
                                by_vars = exprs(USUBJID, FAOBJ, ATPTREF)) {
```

**Review by bms63 — COMMENTED** [review]
This is looking good.  Is it possible to create tests?  Or is that happening in a separate branch?

**Comment by bms63** [comment]
> Hi @bms63, All checks are successful. And I have just updated the derive_param_maxsev_draft.R file with New code. So you can review the updated code and give your comments. Since i have modified the draft code and it's not a draft one anymore, Is it possible to rename the .R file in devel?

You don't need to rename - git is tracking the changes in the file.  So you should be good.  This is another reasons to write unit tests as you develop so we can be more confident the function is doing what we say it is doing. 
image

**Comment by arjoon-r** [comment]
> > Hi @bms63, All checks are successful. And I have just updated the derive_param_maxsev_draft.R file with New code. So you can review the updated code and give your comments. Since i have modified the draft code and it's not a draft one anymore, Is it possible to rename the .R file in devel?
> 
> You don't need to rename - git is tracking the changes in the file. So you should be good. This is another reasons to write unit tests as you develop so we can be more confident the function is doing what we say it is doing. image

Thanks Ben ! Making sense :)

**Pull request merged**